CMake/SetFolder.cmake

```diff
@@ -0,0 +1,33 @@
+function(set_folder directory folder)
+  set(options)
+  set(oneValueArgs)
+  set(multiValueArgs EXCLUDE_TARGET EXCLUDE_DIRECTORY)
+  cmake_parse_arguments(SF "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+
+  set(exclude_directory)
+  foreach(excl IN LISTS SF_EXCLUDE_DIRECTORY)
+    get_filename_component(excl_abs "${excl}" ABSOLUTE)
+    list(APPEND exclude_directory ${excl_abs})
+  endforeach()
+
+  get_directory_property(targets DIRECTORY "${directory}" BUILDSYSTEM_TARGETS)
```

Comment:
CMake 3.7 is required to list all the targets recursively.
Unless we do it manually per target (we will at some point), this is a good start to make it usable.